docker/Dockerfile

```diff
@@ -1,4 +1,34 @@
-FROM ubuntu:xenial
+FROM ubuntu:xenial as builder
```

Comment:
Should we be pinning a version here?

Reply:
I don't think its necessary. Doesn't buy us much. The apt caches are still blown away so we'd still need to do an `apt-get update` for most of the things we want. I've got a jira ticket to possibly explore pinning the version to a date and what not but that'll be left for another day.